android/src/main/java/com/swmansion/gesturehandler/core/HoverGestureHandler.kt

```diff
@@ -0,0 +1,114 @@
+package com.swmansion.gesturehandler.core
+
+import android.os.Handler
+import android.os.Looper
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import com.swmansion.gesturehandler.react.RNViewConfigurationHelper
+
+class HoverGestureHandler : GestureHandler<HoverGestureHandler>() {
+  private var handler: Handler? = null
+  private var finishRunnable = Runnable { finish() }
+
+  private infix fun isAncestorOf(other: GestureHandler<*>): Boolean {
+    var current: View? = other.view
+
+    while (current != null) {
+      if (current == this.view) {
+        return true
+      }
+
+      current = current.parent as? View
+    }
+
+    return false
+  }
+
+  private fun isViewDisplayedOverAnother(view: View, other: View, rootView: View = view.rootView): Boolean? {
+    if (rootView == other) {
+      return true
+    } else if (rootView == view) {
+      return false
+    } else if (rootView is ViewGroup) {
+      for (i in 0 until rootView.childCount) {
+        val child = viewConfigHelper.getChildInDrawingOrderAtIndex(rootView, i)
+        return isViewDisplayedOverAnother(view, other, child) ?: continue
+      }
+    }
+
+    return null
```

Comment:
I wouldn't make if...else chain since the if statements have returns inside

```suggestion
if (rootView == other) {
  return true
} 
if (rootView == view) {
  return false
} 

if (rootView is ViewGroup) {
  for (i in 0 until rootView.childCount) {
    val child = viewConfigHelper.getChildInDrawingOrderAtIndex(rootView, i)
    return isViewDisplayedOverAnother(view, other, child) ? : continue
  }
}

return null
```

Reply:
Same as above, tried using `when` but in this case it might have ended up worse than before (https://github.com/software-mansion/react-native-gesture-handler/pull/2455/commits/1a95954dbff666c337357771e9e989dfb47ee176), what do you think?

Reply:
In that case I think that leaving if's with returns (like in my suggestion) looks a bit clearer. Also the case with `ViewGroup` looks better in if version.

Reply:
In that case I think that leaving if's with returns (like in my suggestion) looks a bit clearer. Also the case with `ViewGroup` looks better in if version.

Reply:
In that case I think that leaving if's with returns (like in my suggestion) looks a bit clearer. Also the case with `ViewGroup` looks better in if version.

Reply:
Agreed - https://github.com/software-mansion/react-native-gesture-handler/pull/2455/commits/c498d0593765d8bfa3482b7f9a0ebb8249517fae